Automattic/wp-calypso
# NUX: Refactor step site topic state

## Changes proposed in this Pull Request

This PR refactors the state tree for the signup step `siteTopic`. Currently we are saving the string value – `state.signup.steps.siteTopic`. This PR introduces a `siteVertical` object:

```
state.signup.steps.siteVertical = {
   name: 'Mexikanisches Restaurant ', // the localized vertical name, formerly `siteTopic` 
   slug: 'Mexican Restaurant', // the slug (English) used for tracking
}
```

### Why do this?

The site verticals API returns [extra vertical metadata](https://public-api.wordpress.com/wpcom/v2/verticals?search=restaurant), some of which we need in our app and therefore needs to be saved in the state. 

Right now, we need at the very least `vertical_slug` (the English language translation of our verticals) so that we can consistently track values. PR #29172, which integrates the vertical API into onboarding, is doing exactly this. 

## Testing instructions
1. Fire up this branch and create a site as per normal on the `onboarding` signup flow
2. Ensure `state.signup.steps.siteVertical.name` is set at the site topic
3. Preserve the HTTP log and create a new site
4. We should be sending the correct vertical option value:
<img width="1183" alt="screen shot 2018-12-24 at 12 28 15 pm" src="https://user-images.githubusercontent.com/6458278/50389014-33521c80-0778-11e9-941f-6aa4e15e363f.png">
